src/qibo/models/circuit.py

```diff
@@ -1267,18 +1268,7 @@ def _update_draw_matrix(self, matrix, idx, gate, gate_symbol=None):
 
         return matrix, idx
 
-    def draw(self, line_wrap=70, legend=False) -> str:
-        """Draw text circuit using unicode symbols.
-
-        Args:
-            line_wrap (int): maximum number of characters per line. This option
-                split the circuit text diagram in chunks of line_wrap characters.
-            legend (bool): If ``True`` prints a legend below the circuit for
-                callbacks and channels. Default is ``False``.
-
-        Return:
-            String containing text circuit diagram.
-        """
+    def __str__(self, line_wrap: int = 70, legend: bool = False) -> str:
```

Comment:
So, sorry for the confusion of my initial proposal, I didn't take into account the arguments.

Let me formulate a complete one here (if needed I could also implement it to speed up the process)
```py
def diagram(self, line_wrap, ...) -> str:
    ...

def __str__(self):
    return self.diagram()

def draw(self, line_wrap, ...) -> str:
    """Just an alias, to avoid breaking compatibility."""
    return self.diagram(line_wrap, ...)

def display(self, line_wrap, ...):
    """Here for compatibility, to be replaced by `draw`."""
    sys.stdout.write(self.diagram(line_wrap, ...)) 
```

Whenever we will decide that it is acceptable to break, we will just drop `display` and replace it with `draw`.

Reply:
Done